Yuneec_SDK_iOS/YNCSDKCamera.mm

```diff
@@ -746,6 +746,48 @@ void receive_metering_result(YNCMeteringCompletion completion, Camera::Result re
     return cameraMeteringMode;
 }
 
+//MARK: receive camera media info result
+void receive_camera_media_result(YNCCameraMediaCompletion completion, Camera::Result result, int progress) {
+    if (completion) {
+        NSError *error = nullptr;
+        if (result != Camera::Result::SUCCESS) {
+            NSString *message = [NSString stringWithFormat:@"%s", Camera::result_str(result)];
+            error = [[NSError alloc] initWithDomain:@"Camera"
+                                               code:(int)result
+                                           userInfo:@{@"message": message}];
+            completion(progress, error);
+        }
+        else {
+            completion(progress, error);
+        }
+    }
+}
+
+//MARK: receive camera all media info result
+void receive_camera_all_media_result(YNCCameraMediaInfosCompletion completion, Camera::Result result, std::vector<Camera::MediaInfo> mediaInfos) {
+    if (completion) {
+        NSError *error = nullptr;
+        if (result != Camera::Result::SUCCESS) {
+            NSString *message = [NSString stringWithFormat:@"%s", Camera::result_str(result)];
+            error = [[NSError alloc] initWithDomain:@"Camera"
+                                               code:(int)result
+                                           userInfo:@{@"message": message}];
+            completion(nil, error);
+        }
+        else {
+            NSMutableArray<YNCCameraMediaInfo *> *YNCCameraMediaInfos = [NSMutableArray new];
```

Comment:
I have newer seen this allocation with `new`.
I would usually do `[[NSMutableArray alloc]init]`.

Also, I wouldn't make an array with pointers to `YNCCameraMediaInfo` but actual struct objects, and then just copy them in there by value.

Reply:
Sorry forgot to change this! I am not sure what you mean. Can you please elaborate?

Reply:
What I meant is that you don't need to allocate the `YNCCameraMediaInfo` on the heap using a pointer and `new` but, I think, you can just create it on the stack:
```
YNCCameraMediaInfo mediaInfo;
mediaInfo.path = @(mediaInfos[i].path.c_str());
mediaInfo.sizeMib = mediaInfos[i].size_mib;
```
and then add it (by value) so it gets copied.

Reply:
Actually, it doesn't work as I suggested, so nevermind.